Update from .NET 4.6 to .NET 4.6.2

koepalex opened the pull request:
Upgraded all project from .NET 4.6 to .NET 4.6.2 that also .NET framework applications can use latest version of Kestrel

* Updated Kestrel to 2.2.0 (for Opc.Ua.Core)
* Simplified csproj (removed ItemGroup Condition="'$(TargetFramework)' == 'netstandard1.3'") because currently we don't build netstandard1.x anymore
* Removed Kestrel 1.1.3 code from Stack/Opc.Ua.Core/Stack/Https/HttpsListener.cs 
* add conditions to Stack/Opc.Ua.Core/Stack/Https/HttpsTransportChannel.cs 
  * [HttpClientHandler.ServerCertificateCustomValidationCallback](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.servercertificatecustomvalidationcallback?view=netcore-3.1&viewFallbackFrom=netframework-4.6.2) should not be available in .NET 4.6.X
* Updated travis.ci to use latest mono version (original version causes build errors)
* Added nuget reference .NET Standard Library to mono projects to be compatible with latest Kestrel version

Own Travis.ci and Azure Pipeline builds run sucessfully.

mregen commented on Stack/Opc.Ua.Core/Opc.Ua.Core.csproj:
```diff
@@ -36,54 +36,17 @@
     <Reference Include="System.IdentityModel.Selectors" />
     <Reference Include="System.Security" />
     <Reference Include="System.ServiceModel" />
+    <Reference Include="System.Net.Http" Version="4.3.0" />
```
><Reference Include="System.Net.Http" Version="4.3.0" /> [](start = 4, length = 55)

does this cause a new warning in .Net projects? Or shouldn't this ref have no version?

koepalex replied:
You are right, it was a leftover to force deal with bug in old mono build (for travis ci), I've removed it that the default version from targetframework will be used.

mregen commented:
> #if NETSTANDARD2_0

can be removed too?

---
Refers to: Stack/Opc.Ua.Core/Stack/Https/HttpsListener.cs:39 in a99c4af. [](commit_id = a99c4af38ddce13a5168a94654965ce58860da3a, deletion_comment = False)

koepalex commented:
> The only NET461 remainings I found are
> https://github.com/OPCFoundation/UA-.NETStandard/blob/master/SampleApplications/Workshop/Reference/Server/App.config#L62
> and
> 
> https://github.com/OPCFoundation/UA-.NETStandard/blob/master/ComIOP/Wrapper/ServerWrapper/App.config#L73
> 
> Otherwise it looks good

I forgot the app.config complety, updated all App.config from 4.6 and 4.6.1 to 4.6.2 also removed the obsolete assembly rebindings

koepalex commented:
> > #if NETSTANDARD2_0
> 
> can be removed too?
> 
> Refers to: Stack/Opc.Ua.Core/Stack/Https/HttpsListener.cs:39 in [a99c4af](https://github.com/OPCFoundation/UA-.NETStandard/commit/a99c4af38ddce13a5168a94654965ce58860da3a). [](commit_id = [a99c4af](https://github.com/OPCFoundation/UA-.NETStandard/commit/a99c4af38ddce13a5168a94654965ce58860da3a), deletion_comment = False)

The #if and the included using could be removed completly, the using was not used at all.
